library/components/TTooltip.vue

```diff
@@ -38,17 +37,15 @@ export default {
 			type: String,
 			default: "top"
 		},
+		isOpen: { // To control state from a component.
+			type: Boolean,
+			default: false,
+		}
 	},
 	data: () => ({
 		positions: {},
-		showTooltip: false,
-        isFocused: false,
+		isVisible: false,
```

Comment:
Can you do this instead and skip the watcher?
```suggestion
		isVisible: isOpen,
```

Reply:
If we use this method, then placement logic might not work and depending on window size and position of elements at that point, tooltip will not be able to align itself properly.

I am sure placement logic can be handled better, though it requires a lot more time I think to change.

Reply:
Watcher makes sure that placement is also triggered on change.

Reply:
Oh wait, sorry I forgot that I changed that, you are right, trying this approach.